src/overlays/actors/ovl_En_Tg/z_en_tg.c

```diff
@@ -76,37 +91,200 @@ static DamageTable D_809901F8 = {
+    this->actionFunc(this, play);
+    Actor_UpdateBgCheckInfo(play, &this->actor, 0.0f, 0.0f, 0.0f, 4U);
+    EnTg_UpdateSkelAnime(this, play);
+    EnTg_UpdateHeartPath(play, &this->enTgHeartInfo, 10);
+    EnTg_UpdateCollider(this, play);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tg/func_8098F8A8.s")
+s32 EnTg_OverrideLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnTg* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tg/func_8098F928.s")
+    return 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tg/EnTg_Init.s")
+void EnTg_PostLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    EnTg* this = THIS;
+    Vec3f zeroVec = { 0.0f, 0.0f, 0.0f };
+
+    if (limbIndex == HONEY_AND_DARLING_LIMB_MAN_HEAD) {
+        Matrix_MultVec3f(&zeroVec, &this->actor.focus.pos);
+    }
+}
+
+void EnTg_Draw(Actor* thisx, PlayState* play) {
+    EnTg* this = THIS;
+    GraphicsContext* gfxCtx;
+
+    Matrix_Push();
+    EnTg_DrawHeart(play, &this->enTgHeartInfo, 10);
+    Matrix_Pop();
+
+    OPEN_DISPS(play->state.gfxCtx);
+    func_8012C28C(play->state.gfxCtx);
+
+    gDPPipeSync(POLY_OPA_DISP++);
+    gSPSegment(POLY_OPA_DISP++, 0x08, Gfx_EnvColor(play->state.gfxCtx, 0, 0x32, 0xA0, 0));
+    gSPSegment(POLY_OPA_DISP++, 0x09, Gfx_EnvColor(play->state.gfxCtx, 0xFF, 0xFF, 0xFF, 0));
+
+    SkelAnime_DrawFlexOpa(play, this->skelAnime.skeleton, this->skelAnime.jointTable, (s32)this->skelAnime.dListCount,
+                          EnTg_OverrideLimbDraw, EnTg_PostLimbDraw, &this->actor);
+
+    CLOSE_DISPS(play->state.gfxCtx);
+}
+
+/**
+ * This function is always called with the same heartStartPos and len = 10.
+ * Sets all the flags and a path for when the first heart (of two) is spawned.
+ */
+void EnTg_SpawnFirstHeart(EnTg* this, EnTgHeartInfo* enTgHeartInfo, Vec3f* heartStartPos, s32 len) {
+    Vec3f heartVelocityVec = { 0.0f, 1.5f, 0.0f };
+    Vec3f zeroVec = { 0.0f, 0.0f, 0.0f };
+    s32 i;
+
+    for (i = 0; i < len && enTgHeartInfo->isFirstHeartSpawned; i++, enTgHeartInfo++) {}
+
+    if (i < len) {
+        enTgHeartInfo->isFirstHeartSpawned = true;
+        enTgHeartInfo->firstHeartPos = *heartStartPos;
+        enTgHeartInfo->velocity = heartVelocityVec;
+        enTgHeartInfo->unusedZeroVec20 = zeroVec;
+        enTgHeartInfo->scale = 0.01f;
+        enTgHeartInfo->firstHeartPos.x += 4.0f * Math_SinS(this->actor.shape.rot.y);
+        enTgHeartInfo->firstHeartPos.z += 4.0f * Math_CosS(this->actor.shape.rot.y);
+        enTgHeartInfo->firstHeartTimer = 16;
+    }
+}
+
+/**
+ * This function is always called with the same len = 10.
+ * The heart path is curvy as it floats up because of the use of Math_SinS and Math_CosS.
+ * The first heart spawned sets the path, the second heart spawned follows it.
+ */
+void EnTg_UpdateHeartPath(PlayState* play, EnTgHeartInfo* enTgHeartInfo, s32 len) {
```

Comment:
```suggestion
void EnTg_UpdateHearts(PlayState* play, EnTgHeartInfo* enTgHeartInfo, s32 len) {
```
It seems to be updating everything regarding the hearts.